staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -362,8 +362,10 @@ func (e *Store) WaitForInitialized(ctx genericapirequest.Context, obj runtime.Ob
 		select {
 		case event, ok := <-ch:
 			if !ok {
-				// TODO: should we just expose the partially initialized object?
-				return nil, kubeerr.NewServerTimeout(e.QualifiedResource, "create", 0)
+				return latest,
```

Comment:
I already did!  https://github.com/kubernetes/kubernetes/pull/49678 correctly returns 504 gateway timeout from the timeout handler, which should be `IsTimeout`, which if you return it here gives you an indication that the server was unable to complete your op, you should wait and see if it goes through.